examples/batch_calls.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019-2021 Parity Technologies (UK) Ltd.
+// This file is part of subxt.
+//
+// subxt is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// subxt is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with subxt.  If not, see <http://www.gnu.org/licenses/>.
+
+//! To run this example, a local polkadot node should be running.
+//!
+//! E.g.
+//! ```bash
+//! curl "https://github.com/paritytech/polkadot/releases/download/v0.9.11/polkadot" --output /usr/local/bin/polkadot --location
+//! polkadot --dev --tmp
+//! ```
+
+use sp_keyring::AccountKeyring;
+use subxt::{ClientBuilder, PairSigner};
+
+#[subxt::subxt(runtime_metadata_path = "examples/polkadot_metadata.scale")]
+pub mod polkadot {}
+
+type Call = polkadot::runtime_types::polkadot_runtime::Call;
+type BalancesCall = polkadot::runtime_types::pallet_balances::pallet::Call;
+
+#[async_std::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    env_logger::init();
+
+    let alice = PairSigner::new(AccountKeyring::Alice.pair());
+    let bob = AccountKeyring::Bob.to_account_id().into();
+    let charlie = AccountKeyring::Charlie.to_account_id().into();
+
+    let api = ClientBuilder::new()
+        .build()
+        .await?
+        .to_runtime_api::<polkadot::RuntimeApi<polkadot::DefaultConfig>>();
+
+    let calls = vec![
+        Call::Balances(BalancesCall::transfer {
```

Comment:
> It's worth noting that this depends on the target runtime pallets having the same indices as that of the static metadata used to generate the API. When running a single call we dynamically add the index prefix based on the metadata fetched at runtime.
> 
> We probably need some mechanism to compare the static metadata used to generate the API with the metadata fetched from the node at runtime.

@ascjones I assume the same is true for the `sudo` wrapper? With both of these calls, I get `panicked at Bad input data provided to validate_transaction: Codec error`. Is this the error you would expect when there is an index mismatch? The strange thing is, though, that I get this error despite the fact the both the static and dynamic metadata are generated by the same binary

Reply:
Yes we would, it means the call can't be decoded which means we are encoding it wrong. I am vaguely aware that `sudo` calls are handled slightly differently, so the autogenerated types might not work for this pallet. It needs looking into - I won't have time this week since I'm away. A clue is to look how it was handled here previously, with a raw encoded call: https://github.com/paritytech/subxt/blob/master/src/frame/sudo.rs#L37